### ISSUE TITLE
+RTS .. -RTS sections in ghc-options shouldn't affect the artifact hash

### ISSUE THREAD

**Issue by 23Skidoo** [haskell/cabal#5575]
If I build a project with `new-build` and then do `new-clean && new-build --ghc-options="+RTS -A32M"`, it will try to rebuild all dependencies as well. This is not necessary and shouldn't happen.

**Comment by 23Skidoo** [comment]
Related: #4247.

/cc @merijn

**Comment by merijn** [comment]
Actually, I don't think this is a matter of "RTS options shouldn't affect the hash" as, for executables, they definitely *do* affect the build output. This actually seems more related to the matter is discussed with @ezyang at some point on whether commandline flags (like `--ghc-options`) should by default be scoped to everything or just the current package (my argument then and in some ticket I can't find) was that commandline flags should *default* to only the current project and not all dependencies.

I'm pretty sure there's already a feature request for this somewhere and that'd obsolete the entire difficult task of somehow distinguishing *some* RTS options as safe and others as not.

**Comment by 23Skidoo** [comment]
> Actually, I don't think this is a matter of "RTS options shouldn't affect the hash" as, for executables, they definitely *do* affect the build output.

Hmm, I don't see how they do. This is about RTS options passed to the GHC executable itself via `+RTS` (e.g. to make the GHC process use a bigger GC nursery during compilation), not `-rtsopts` (default RTS options for the resulting exes).

Though if `-rtsopts` affects only exes and not libraries (I think it affects foreign libraries as well), it'd be nice if it didn't cause libraries to be rebuilt.

**Comment by merijn** [comment]
oh, I misinterpreted the problem! This should actually be fairly trivial. It's simply a matter of modifying normaliseArgs (https://github.com/haskell/cabal/blob/86dabda8ff1eb31e80dc1ca2eff21e2839211fdd/Cabal/Distribution/Simple/Program/GHC.hs#L49) to strip out everything from +RTS to -RTS. But I don't have time to implement that right now.

If anyone else takes it up they should dive into the GHC 8.6 code and check if any flags were added/changed while they're at it, so the supported GHC range can be expanded to include 8.6.1.
